sklearn/tree/_tree.pyx

```diff
@@ -531,13 +702,19 @@ cdef class Gini(ClassificationCriterion):
             gini_right = 0.0
 
             for c in range(n_classes[k]):
-                tmp = label_count_left[c]
-                gini_left += tmp * tmp
-                tmp = label_count_right[c]
-                gini_right += tmp * tmp
+                # Add to the Gini index of the left node
+                count_k = label_count_left[c]
+                gini_left += count_k * count_k
 
+                # Add to the Gini index of the right node
+                count_k = label_count_right[c]
+                gini_right += count_k * count_k
+
+            # Normalize the Gini index of the left node
```

Comment:
This is incorrect. Strictly speaking, at this point, `gini_left` is not yet a gini index per se. It is only after that line that it becomes one. 

I would simply remove this comment.